### PR TITLE
ansible: introduce skydive library

### PR DESCRIPTION
**Pull request by safchain**
This exposes the skydive API through ansible

### CODE REVIEW

File: contrib/ansible/library/skydive_edge.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+       
+        return str(nodes[0].id)
+
+    return node_selector
+
+
+def run_module():
+    module_args = dict(
+        analyzer=dict(type='str', default="localhost:8082"),
+        ssl=dict(type=bool, default=False),
```
Comment (lebauce) [inline contrib/ansible/library/skydive_edge.py]: `type=bool` need to be replaced by `type='bool'`
and
`type=str` by `type='str'`

File: contrib/ansible/library/skydive_edge.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+                            username=module.params["username"], password=module.params["password"],
+                            module=module,
+                            params=module.params,
+                            node1=node1,
+                            node2=node2,
+                            result=result)
+        wsclient.connect()
+        wsclient.start()
+    except Exception as e:
+        module.fail_json(msg='Connection error %s' % e.message, **result)
```
Comment (lebauce) [inline contrib/ansible/library/skydive_edge.py]: On my setup, there was nothing in `message` so I did `str(e)` instead

**Comment by safchain** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
run skydive-scale-tests

**Comment by lebauce** [comment]
run skydive-scale-tests
run skydive-k8s-tests
run skydive-unit-tests

File: contrib/ansible/library/skydive_node.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+            host = params["host"]
+
+            metadata = params["metadata"]
+            metadata["Name"] = params["name"]
+            metadata["Type"] = params["type"]
+
+            sid = params["id"]
+            if len(sid) == 0:
+                sid = "%s:%s" % (params["name"], params["type"])
+            uid = str(uuid.uuid5(uuid.NAMESPACE_OID, sid))
```
Comment (lebauce) [inline contrib/ansible/library/skydive_node.py]: Remove leading space

File: contrib/ansible/library/skydive_node.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+            metadata = params["metadata"]
+            metadata["Name"] = params["name"]
+            metadata["Type"] = params["type"]
+
+            sid = params["id"]
+            if len(sid) == 0:
+                sid = "%s:%s" % (params["name"], params["type"])
+            uid = str(uuid.uuid5(uuid.NAMESPACE_OID, sid)) 
+
```
Comment (lebauce) [inline contrib/ansible/library/skydive_node.py]: nit: remove whitespaces

File: contrib/ansible/library/skydive_node.py [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+    result = dict(
+        changed=False
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
```
Comment (lebauce) [inline contrib/ansible/library/skydive_node.py]: nit: remove whitespaces

File: contrib/ansible/library/skydive_edge.py [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+        restclient = RESTClient(module.params["analyzer"], scheme=scheme,
+                                insecure=module.params["insecure"],
+                                username=module.params["username"],
+                                password=module.params["password"])
+        nodes = restclient.lookup_nodes(node_selector)
+        if len(nodes) == 0:
+            raise Exception("Node not found: %s" % node_selector)
+        elif len(nodes) > 1:
+            raise Exception("Node selection should return only one node: %s" % node_selector)
+
```
Comment (lebauce) [inline contrib/ansible/library/skydive_edge.py]: nit: remove whitespaces

**Pull request merged**